nle/env/base.py

```diff
@@ -216,6 +216,7 @@ def __init__(
         allow_all_modes=False,
         space_dict=None,
         spawn_monsters=True,
+        episode_save_cycle=1,
```

Comment:
While I like this idea, I think this is also a good opportunity to rethink this API a bit.

We've not done some changes to this API that we wanted to do in the past (e.g., remove `archivefile` which doesn't do anything anymore) in order to not create undue churn during the NetHack competition.

After the competition, we have another chance getting some of the behaviors right. I could imagine:

- Calling this `save_episode_every`, or even `episodes_per_ttyrec` (there might be something better).
- Right now, `savedir=None` disables saving of episodes. `savedir=''` is a magic argument that creates a random temporal directory. Any other string is taken as a path. That is a bit too magical. With this new argument, we could turn `0` into "don't save at all", which would free `savedir=None` to mean what `savedir=''` does right now.

But we'll need a bit of time to discuss between @condnsdmatters, @tscmoo, @samvelyan and myself.

Reply:
Yeah - this is a *lot* faster, so internally we're working off this branch. If you guys have a slack channel (with public sub-channels) happy to weigh in or continue off our internal branch

Reply:
Not saving any ttyrecs is _even_ faster ;)

But I get that having _some_ ttyrecs is useful, too.

We don't currently have slack channel, but you can reach us on the NetHack competition discord. Note though that the competition runs until mid October and we are probably not going to make API changes until then -- would you be okay with working on your own branch until then?

Reply:
Yeah - Im fine with that (@deanpfoster)